pkg/kvstore/kvstore.go

```diff
@@ -44,13 +38,94 @@ type KVClient interface {
 	GetWatcher(key string, timeSleep time.Duration) <-chan []policy.NumericIdentity
 
 	Status() (string, error)
+
+	// Get returns value of key
+	Get(key string) ([]byte, error)
+
+	// GetPrefix returns the first key which matches the prefix
+	GetPrefix(prefix string) ([]byte, error)
+
+	// Set sets value of key
+	Set(key string, value []byte) error
+
+	// Delete deletes a key
+	Delete(key string) error
+
+	// CreateOnly atomically creates a key or fails if it already exists
+	CreateOnly(key string, value []byte, lease bool) error
+
+	// CreateIfExists creates a key with the value only if key condKey exists
+	CreateIfExists(condKey, key string, value []byte, lease bool) error
+
+	// ListPrefix returns a list of keys matching the prefix
+	ListPrefix(prefix string) (KeyValuePairs, error)
+
+	// StartWatch starts watching for changes in a prefix
+	StartWatch(w *Watcher)
```

Comment:
What's the benefit of this? It will require me to duplicate the code that's currently in `events.go` into both etcd and consul. I may misunderstand something there.